### PR TITLE
RGW: Solving the issue of not populating etag in Multipart upload result

### PR DESCRIPTION
**Pull request by AliMasarweh**
Solving the issue of not populating etag in Multipart upload result
https://tracker.ceph.com/issues/58879

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/rgw/rgw_op.cc [hunk 1]
```diff
@@ -6575,6 +6575,9 @@ void RGWCompleteMultipart::complete()
       ldpp_dout(this, 0) << "WARNING: failed to unlock " << *serializer.get() << dendl;
     }
   }
+
+  etag = s->object->get_attrs()[RGW_ATTR_ETAG].to_str();
```
Comment (mattbenjamin) [inline src/rgw/rgw_op.cc:6579]: that has a plausible feel to it ;)

**Comment by dang** [comment]
I forgot to mention that all Ceph commits need to be signed off.  This is done with the "-s" option to "git commit".  You  can fix this in this commit with:
git commit -s --amend
And then save the commit message unchanged.  Then force push ("git push -f") the branch again.

**Comment by AliMasarweh** [comment]
Signed the commit and pushed again

**Review by dang — APPROVED** [review]

**Comment by cbodley** [comment]
there's a qa/workunits/rgw/run-s3tests.sh script that you can run from your ceph build directory. it will clone the https://github.com/ceph/s3-tests repo and run the subset of test cases that we expect to pass

you can add a test case for this fix to https://github.com/ceph/s3-tests/blob/master/s3tests_boto3/functional/test_s3.py

**Comment by alimaredia** [comment]
@cbodley in teuthology should @AliMasarweh just be running just the rgw:verify suite with a revised s3-tests for this fix?

**Comment by cbodley** [comment]
> @cbodley in teuthology should @AliMasarweh just be running just the rgw:verify suite with a revised s3-tests for this fix?

the full rgw suite against revised s3-tests please

**Comment by cbodley** [comment]
@AliMasarweh can you please share the teuthology results?

**Comment by cbodley** [comment]
and is there an associated s3test case that needs to merge also?

**Comment by AliMasarweh** [comment]
this is the teuthology results
http://pulpito.front.sepia.ceph.com/alimasa-2023-04-04_09:48:28-rgw-wip-ali-masa-multipart-populate-etag-distro-default-smithi/
I haven't touched any of the s3test cases, maybe @dang can tell if we need to add a test to verify we populate the etag

**Pull request merged**